modules/host/host.go

```diff
@@ -30,6 +30,21 @@ package host
 // TODO: 'announced' doesn't tell you if the announcement made it to the
 // blockchain.
 
+// TODO: Need to make sure that the revision exchange for the renter and the
+// host is being handled correctly. For the host, it's not so difficult. The
+// host need only send the most recent revision every time. But, the host
+// should not sign a revision unless the renter has explicitly signed such that
+// the 'WholeTransaction' fields cover only the revision and that the
+// signatures for the revision don't depend on anything else. The renter needs
+// to verify the same when checking on a file contract revision from the host.
+// If the host has submitted a file contract revision where the signatures have
+// signed the whole file contract, there is an issue.
+
+// TODO: there is a mistake in the file contract revision rpc, the host, if it
+// does not have the right file contract id, should be returning an error there
+// to the renter (and not just to it's calling function without informing the
+// renter what's up).
```

Comment:
these todos mostly serve as triggers for me to remember to check various things. In this case, I don't think the contents are 100% relevant anymore as this PR solves some of the things listed, but I still haven't done all the actions mentioned. Improvements will be ongoing.